examples/colors/color.rs

```diff
@@ -0,0 +1,194 @@
+                    ParamSpec::new_float(
+                        "green",
+                        "green",
+                        "green",
+                        f32::MIN,
+                        f32::MAX,
+                        0.,
+                        ParamFlags::READWRITE,
+                    ),
+                    ParamSpec::new_float(
+                        "blue",
+                        "blue",
+                        "blue",
+                        f32::MIN,
+                        f32::MAX,
+                        0.,
+                        ParamFlags::READWRITE,
+                    ),
+                    ParamSpec::new_float(
+                        "hue",
+                        "hue",
+                        "hue",
+                        f32::MIN,
+                        f32::MAX,
+                        0.,
+                        ParamFlags::READWRITE,
+                    ),
+                    ParamSpec::new_float(
+                        "saturation",
+                        "saturation",
+                        "saturation",
+                        f32::MIN,
+                        f32::MAX,
+                        0.,
+                        ParamFlags::READWRITE,
+                    ),
+                    ParamSpec::new_float(
+                        "value",
+                        "value",
+                        "value",
+                        f32::MIN,
+                        f32::MAX,
+                        0.,
+                        ParamFlags::READWRITE,
+                    ),
+                ]
+            });
+            PROPERTIES.as_ref()
+        }
+
+        fn set_property(&self, _obj: &Self::Type, _id: usize, value: &Value, pspec: &ParamSpec) {
+            match pspec.name() {
+                "color" => {
+                    let rgba = value.get().expect("The value needs to be of type `i32`.");
+                    self.color.replace(rgba);
+
+                    self.red.replace(rgba.red());
+                    self.green.replace(rgba.green());
+                    self.blue.replace(rgba.blue());
+
+                    let (h, s, v) = gtk::rgb_to_hsv(rgba.red(), rgba.green(), rgba.blue());
+                    self.h.replace(h);
+                    self.s.replace(s);
+                    self.v.replace(v);
+                }
+                _ => unimplemented!(),
+            }
+        }
+
+        fn property(&self, _obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> Value {
+            match pspec.name() {
+                "name" => self.name.borrow().to_value(),
+                "color" => self.color.get().to_value(),
+                "red" => self.red.get().to_value(),
+                "green" => self.green.get().to_value(),
+                "blue" => self.blue.get().to_value(),
+                "hue" => self.h.get().to_value(),
+                "saturation" => self.s.get().to_value(),
+                "value" => self.v.get().to_value(),
```

Comment:
do you need any of those properties other than color?

Reply:
yes. for sorting